### ISSUE TITLE
Add a maps component

### ISSUE THREAD

**Issue by Dev1an** [Quite-nice/Visualisation#21]

**Comment by jeroenbe** [comment]
After several attempts spread over the last view days, I'm running out of options. Some help would be greatly appreciated

**Comment by Dev1an** [comment]
I'm first going to close #11 and #22, after that I will look into it.

**Comment by Dev1an** [comment]
@JeroenBe: Have you tried to wrap the code inside a meteor 1.3 module? Did you ask the mapbox/leaflet/google-maps community for some help?

**Comment by jeroenbe** [comment]
There is quite some community chatter about the issue, both on the [Leaflet github page](https://github.com/Leaflet/Leaflet/issues/3229) as in the [npm community surrounding Mapbox](https://www.npmjs.com/package/mapbox-gl). Even dedicated [Atmosphere packages](https://github.com/pauloborges/meteor-mapbox/issues/22) concerning mapbox receive a fair share of _complaints_.

I browsed to pretty much all of them and tried everything they suggested, without luck. I asked around on a newly created [thread](https://forums.meteor.com/t/having-issues-importing-mapbox-gl-with-1-3/21454) on the meteor forums, here someone suggested using a script tag with the actual mapbox-gl.js files. Up un till now, this is they only suggestion that works. Sadly I find it a rather dirty one, but pragmatically, it should suffice for now. 

My next step would be to include my findings in the Visualisation project.

**Comment by jeroenbe** [comment]
### Concerning Access Tokens

After reading the[ explanation of the mapbox access tokens](https://www.mapbox.com/help/define-access-token/), I feel tempted to just put them in the Template code instead of placing them in a settings file on startup. It would be great if I could here some opinions on the matter.

FYI, the access token I generated to my account starts with `pk`, as explained on the mapbox documentation.

@Dev1an @Kymer @maccradar

**Comment by jeroenbe** [comment]
It appears FlowRouter is causing some troubles with the script tag. The variable `mapboxgl` can not be found in the `map` template. While it was perfectly found in my test project. I will have to look into resolving this.

**Comment by maccradar** [comment]
Have you tried to add it to "onRendered"? The meteor script is typically run before the other APIs, such as google maps, leaflet, mapbox, are loaded, so they recommend to put your code in a Template.onRendered: see [Template.rendered](http://docs.meteor.com/api/templates.html#Template-onRendered).

**Comment by jeroenbe** [comment]
@maccradar Yes all code has been placed in `onRendered`. I created a [repository](https://github.com/Quite-nice/meteor-mapbox-test) in order to do some experiments. Whenever the code related to mapbox is placed in the `/client` folder (outside the `/imports` folder), everything appears to work. But when I need to `import` bits of code (following the `ES6` standard), I seem to get into trouble.

**Comment by jeroenbe** [comment]
I will try to place all mapbox related files inside an Atmosphere package. See how things go from there on out...

**Comment by maccradar** [comment]
Just confirmed the same. Running it in client works "out of the box". I suggest you leave it there and leave the issue open. Unless you cannot proceed when the code is not imported.

On 13 Jun 2016, at 10:41, Jeroen <notifications@github.com<mailto:notifications@github.com>> wrote:

@maccradarhttps://github.com/maccradar Yes all code has been placedin onRendered. I created a repository in order to do some experiments. Whenever the code related to mapbox is placed in the /client folder (outside the /imports folder), everything appears to work. But when I need to import bits of code (following the ES6 standard), I seem to get into trouble.

##

**Comment by Kymer** [comment]
> I feel tempted to just put them in the Template code instead of placing them in a settings file on startup. It would be great if I could here some opinions on the matter.

Be careful when adding access tokens in public code files. There are crawlers that look specifically for oauth tokens of various API vendors, which can then be abused in some form. Your account could get banned from accessing the service if your amount of requests pass a certain threshold for example.

I'm not really sure what the best practices are concerning API access tokens in open-source projects... maybe others can shed some light on that? @Dev1an @maccradar
